django/db/migrations/operations/models.py

```diff
@@ -811,14 +815,18 @@ def state_forwards(self, app_label, state):
         model_state.options[self.option_name] = [idx for idx in indexes if idx.name != self.name]
 
     def database_forwards(self, app_label, schema_editor, from_state, to_state):
-        from_model_state = from_state.models[app_label, self.model_name_lower]
-        index = from_model_state.get_index_by_name(self.name)
-        schema_editor.remove_index(index)
+        model = from_state.apps.get_model(app_label, self.model_name)
```

Comment:
`AddIndex.database_forwards/backwards()` use `to_state` / `from_state` respectively, while the same methods on `RemoveIndex` both use `from_state`. If it's correct, can you explain why?